### PR TITLE
[Communication] - SMS - Managed Identity Support for SMS Client

### PR DESCRIPTION
**Pull request by minnieliu**
Draft for now: Blocked on running the Live tests for Managed Identity support for SMS. Waiting for service team to enable this functionality and should be available after Jan 15th.

SMS also has some additional complexity because it uses static resources in the live tests. For now, I have skipped the implementation of new live tests for managed identity until we have a solution going forward.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Sms/README.md [hunk 1]
```diff
@@ -36,6 +36,14 @@ string connectionString = "YOUR_CONNECTION_STRING"; // Find your Communication S
 SmsClient client = new SmsClient(connectionString);
 ```
 
+Alternatively, SMS clients can also be authenticated using a valid token.
```
Comment (jbeauregardb) [inline sdk/communication/Azure.Communication.Sms/README.md:39]: Should we tell users to have their client id, tenant and client secret as env variables for DefaultAzureCredential to work?

**Review by jbeauregardb — COMMENTED** [review]
It'd be cool to also add Josh as a reviewer for this one :)

**Review by JoshuaLai — APPROVED** [review]
lgtm

File: sdk/communication/Azure.Communication.Sms/README.md (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/Azure.Communication.Sms/README.md]: Done, thanks for the suggestion Jorge!

File: sdk/communication/Azure.Communication.Sms/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.0.0-beta.4 (Unreleased)
 
+###Added
+- Support for creating PhoneNumberAdministrationClient with TokenCredential
```
Comment (JoshuaLai) [inline sdk/communication/Azure.Communication.Sms/CHANGELOG.md:6]: I think you meant sms here

Comment (minnieliu) [reply sdk/communication/Azure.Communication.Sms/CHANGELOG.md]: Great catch!! Thank you, I've fixed this

**Review by ankitarorabit — APPROVED** [review]
For live tests, sync up with Jorge.. my understanding is that we can still record and do the playback tests while keeping the live tests disabled.. I have asked him to schedule a follow-up to brainstorm the possible options for eventually enabling them

**Comment by minnieliu** [comment]
> For live tests, sync up with Jorge.. my understanding is that we can still record and do the playback tests while keeping the live tests disabled.. I have asked him to schedule a follow-up to brainstorm the possible options for eventually enabling them

@ankitarorabit Just checked with Jorge. We cannot record tests for AAD right now because that Prague team is blocking us. We will be able to record and do playback tests after we are unblocked.

**Review by jbeauregardb — APPROVED** [review]
So cool to see this already working on C# as well. Thank you!

**Pull request merged**